test/quoteSpec.js

```diff
@@ -48,8 +48,130 @@ describe('Quotes', function () {
         .expect(400)
         .expect(function (res) {
           expect(res.body.id).to.equal('NoAmountSpecifiedError')
-          expect(res.body.message).to.equal('Must specify either source or ' +
-            'destination amount to get quote')
+          expect(res.body.message).to.equal('Exactly one of source_amount ' +
+            'or destination_amount must be specified')
+        })
+        .end()
+    })
+
+    it('should return a 400 if both source_amount and destination_amount are specified', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'source_amount=100&destination_amount=100source_ledger=http://eur-ledger.example/EUR' +
+          '&destination_ledger=http://usd-ledger.example/USD')
+        .expect(400)
+        .expect(function (res) {
+          expect(res.body.id).to.equal('InvalidUriParameterError')
+          expect(res.body.message).to.equal('Exactly one of source_amount ' +
+            'or destination_amount must be specified')
+        })
+        .end()
+    })
+
+    it('should return a 400 if source_amount is zero', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'source_amount=0&source_ledger=http://eur-ledger.example/EUR' +
+          '&destination_ledger=http://usd-ledger.example/USD')
+        .expect(400)
+        .expect(function (res) {
+          expect(res.body.id).to.equal('InvalidAmountSpecifiedError')
+          expect(res.body.message).to.equal('source_amount must be finite and positive')
+        })
+        .end()
+    })
+
+    it('should return a 400 if source_amount is infinite', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'source_amount=1e100000&source_ledger=http://eur-ledger.example/EUR' +
+          '&destination_ledger=http://usd-ledger.example/USD')
+        .expect(400)
+        .expect(function (res) {
+          expect(res.body.id).to.equal('InvalidAmountSpecifiedError')
+          expect(res.body.message).to.equal('source_amount must be finite and positive')
+        })
+        .end()
+    })
+
+    it('should return a 400 if destination_amount is zero', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'destination_amount=0&source_ledger=http://eur-ledger.example/EUR' +
+          '&destination_ledger=http://usd-ledger.example/USD')
+        .expect(400)
+        .expect(function (res) {
+          expect(res.body.id).to.equal('InvalidAmountSpecifiedError')
+          expect(res.body.message).to.equal('destination_amount must be finite and positive')
+        })
+        .end()
+    })
+
+    it('should return a 400 if destination_amount is infinite', function * () {
+      yield this.request()
+        .get('/quote?' +
+          'destination_amount=1e100000&source_ledger=http://eur-ledger.example/EUR' +
```

Comment:
Also, `1e100000` is not Infinity? Why would the "Insufficient liqudity" error be incorrect here?